aten/src/ATen/native/Sorting.cpp

```diff
@@ -94,6 +91,169 @@ void quick_select_template(
   } while (true);
 }
 
+// Returns pointer to pivot used to partition range [first, last]
+// There should be at least 3 elements in [first, last]
+template <typename random_it, typename comp>
+random_it partition(random_it first, random_it last, comp lt) {
+  // Use median of three for pivot choice
+  random_it pivot = first + (std::distance(first, last) >> 1);
+  std::swap(first[1], *pivot);
```

Comment:
Would you add a comment here describing what this block does (what invariant is it setting up?)